src/khoj/routers/helpers.py

```diff
@@ -128,15 +141,50 @@ async def agenerate_chat_response(*args):
     return await loop.run_in_executor(executor, generate_chat_response, *args)
 
 
+async def aget_relevant_tools(query: str, conversation_history: dict):
+    """
+    Given a query, determine which of the available tools the agent should use in order to answer appropriately.
+    """
```

Comment:
Should we reframe this more specifically, as get information from relevant data sources (tools sounds too generic IMO)

```suggestion
async def aget_relevant_data_sources(query: str, conversation_history: dict):
    """
    Given a query, determine which of the available data sources the agent should use in order to answer appropriately.
    """
```